src/api.js

```diff
@@ -10,9 +11,18 @@ class Api {
             method,
             body,
             credentials: 'include',
-        }).then((res) => {
+            redirect: 'manual',
+        }).then(async (res) => {
+            if (res.type === 'opaqueredirect') {
+                throw {
+                    message: i18n.t(
+                        'Your session has expired. Please refresh the page and login before trying again.'
+                    ),
+                }
+            }
             if (res.status < 200 || res.status >= 300) {
-                throw new Error(res.statusText)
+                const errorBody = await res.json()
```

Comment:
Is it guaranteed that we'll get a json response for non-200 responses? I'd expect 404 to be `text/plain`.
In that case this threw an error, the error message would be about not being able to parse json rather than the actual issue.